ion/textwriter.go

```diff
@@ -277,47 +288,105 @@ func (w *textWriter) writeValue(api string, val string) error {
 // annotations (if any).
 func (w *textWriter) beginValue(api string) error {
 	if w.needsSeparator {
-		var sep byte
-		switch w.ctx.peek() {
-		case ctxInStruct, ctxInList:
-			sep = ','
-		case ctxInSexp:
-			sep = ' '
-		default:
-			sep = '\n'
+		if err := w.writeSeparator(); err != nil {
+			return err
 		}
+	}
 
-		if err := writeRawChar(sep, w.out); err != nil {
+	if w.emptyContainer {
+		if w.pretty() {
+			if err := writeRawChar('\n', w.out); err != nil {
+				return err
+			}
+		}
+	}
+
+	if w.pretty() {
+		if err := w.writeIndent(); err != nil {
 			return err
 		}
 	}
 
 	if w.inStruct() {
-		if w.fieldName == "" {
-			return &UsageError{api, "field name not set"}
+		if err := w.writeFieldName(api); err != nil {
+			return err
 		}
-		name := w.fieldName
-		w.fieldName = ""
+	}
 
-		if err := writeSymbol(name, w.out); err != nil {
+	if len(w.annotations) > 0 {
+		if err := w.writeAnnotations(); err != nil {
 			return err
 		}
```

Comment:
`writeAnnotations()` graps the annotations from the writer (which is an array) and ranges over them. I think we can just call `writeAnnotations()` and the case when the annotations are empty will output nothing. Or is my newly found Go Lang knowledge failing me here?

Reply:
Yeah it would be safe to always call through to `writeAnnotations`. It reads a bit clearer for me like this ("if there are any annotations, write them out"), but I can change it if you prefer.